drivers/scheduler/dcos/dcos.go

```diff
@@ -106,9 +106,43 @@ func (d *dcos) IsNodeReady(n node.Node) error {
 	return nil
 }
 
-func (d *dcos) GetNodesForApp(*scheduler.Context) ([]node.Node, error) {
-	// TODO: Implement this method
-	return nil, nil
+func (d *dcos) GetNodesForApp(ctx *scheduler.Context) ([]node.Node, error) {
+	var tasks []marathon.Task
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*marathon.Application); ok {
```

Comment:
We handle the object check in the parseSpecs itself. Currently we allow creating only Application object. We followed the same model for k8s scheduler driver because din't want to check it every time we iterate on the speclist. Especially when we know other objects will not seep in.

Are you asking for it so that we can use these functions directly from other projects? If yes, we may have to change it for k8s driver as well, which is slightly tricky.

Reply:
My comment was more to print a warning in case the spec isn't of type marathon.Application.
If that is never expected here (because we've already validated the specs), there should be a log.Fatal/Assert if !ok.

Reply:
Added a warning for now, because we are not exiting the execution if we do not recognize the spec.